packages/contracts/test/LibExperience.t.sol

```diff
@@ -27,4 +33,56 @@ contract LibExperienceTest is MudV2Test {
       assertEq(exp[i], 1);
     }
   }
-}
\ No newline at end of file
+
+  function testGetExp() public {
+    uint32[PStat_length] memory exp = [uint32(10), 20, 30];
+    Experience.set(targetEntity, exp);
+
+    uint32[PStat_length] memory retrievedExp = LibExperience.getExp(targetEntity);
+    for (uint256 i; i < retrievedExp.length; i++) {
+      assertEq(retrievedExp[i], exp[i]);
+    }
+  }
+
+  function testAddExp() public {
+    LibExperience.initExp(targetEntity);
+
+    uint32[PStat_length] memory initialExp = LibExperience.getExp(targetEntity);
+
+    uint32[PStat_length] memory addExp = [uint32(5), 10, 15];
+    LibExperience.increaseExp(targetEntity, addExp);
+
+    uint32[PStat_length] memory expectedExp;
+    for (uint256 i; i < initialExp.length; i++) {
+      expectedExp[i] = initialExp[i] + addExp[i];
+    }
+
+    uint32[PStat_length] memory retrievedExp = LibExperience.getExp(targetEntity);
+    for (uint256 i; i < retrievedExp.length; i++) {
+      assertEq(retrievedExp[i], expectedExp[i]);
+    }
+  }
+
+  function testMaxLevel() public {
+    LibExperience.initExp(targetEntity);
+
+    // Set the experience to the maximum level
+    uint32[PStat_length] memory maxExp = [type(uint32).max, type(uint32).max, type(uint32).max];
+    Experience.set(targetEntity, maxExp);
+
+    assertEq(LibExperience.getAggregateLevel(targetEntity, maxExp), 16);
+  }
+
+  function fuzzTestPStats(uint32[PStat_length] memory pstats) external {
+    // Get expected exp for the random pstat values
+    uint32[PStat_length] memory expectedExp = LibExperience.getExpForPStats(pstats);
+
+    // Get the actual exp
+    uint32[PStat_length] memory actualExp = LibExperience.getExp(targetEntity);
+
+    // Ensure that the actual exp matches the expected exp
+    for (uint256 i = 0; i < expectedExp.length; i++) {
+      assert(actualExp[i] == expectedExp[i]);
+    }
+  }
```

Comment:
1. only functions that start with `test` get run. `fuzz` doesn't have any magic meaning. `testFuzzPStats` or `test_fuzz_PStats` would work
2. exp is neither initialized, nor set
3. never use `assert`, it doesn't do what you think it does. Here `assertEq` would work
4. these functions should be `public`